docs/upgrading/via-dao-vote.md

```diff
@@ -0,0 +1,20 @@
+---
+sidebar_position: 3
+---
+
+# DAO-Governed Updates
+
+When you first deploy a contract to [mainnet](https://docs.near.org/docs/concepts/networks), you will likely keep control of a [Full Access key](https://docs.near.org/docs/concepts/account#access-keys) for the contract. This puts the contract in "trusted" mode, in which you and other maintainers can change it at-will (which means your users need to trust you to not steal their funds, change their votes, or otherwise behave maliciously). This is fine for early-stage contracts & apps, but like any blockchain, NEAR allows you to do better.
+
+When you're ready, you can remove all Full Access keys controlled by individuals, and give only another contract full access to its code _[TODO: is this the way it works? Alternatively, can you remove ALL Full Access keys?]_
```

Comment:
plz advise

Reply:
Yes, you can remove all full access keys, and yes, there is a public function that calls a method that will do the deploy "to itself." Presumably, the best way to do this is to check that the predecessor is indeed, say, a DAO account you trust, indicating that a vote has passed.